utils/parse.js

```diff
@@ -22,6 +22,10 @@ exports.default = function parse(path, content, context) {
   // always attach comments
   parserOptions.attachComment = true
 
+  // provide the `filePath` like eslint itself does, in `parserOptions`
+  // https://github.com/eslint/eslint/blob/3ec436ee/lib/linter.js#L637
+  parserOptions.filePath = path
```

Comment:
The internal parser has a cache that is a hash of the parse settings and the module path. So I'm guessing the path must be relative. 

An absolute path would work, or the cache key could drop the file path key. 

I am not at a computer right now but I think the cache access is in utils/resolve.js